src/Select/Select.jsx

```diff
@@ -105,6 +111,8 @@ Select.propTypes = {
   placeholder: React.PropTypes.string,
   readOnly: React.PropTypes.bool,
   searchable: React.PropTypes.bool,
+  creatable: React.PropTypes.bool,
+  creatableProps: React.PropTypes.object,
```

Comment:
It's hard for me to know what creatable props we want to support. The fact that react-select decided to implement this behavior with a separate component makes me think that they probably did that in order to avoid having a ton of props that only applied in the "creatable" case. I was basically trying to follow their design pattern without actually making a new component.